library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowLongClickListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.applandeo.materialcalendarview.listeners
+
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemLongClickListener
+import com.applandeo.materialcalendarview.EventDay
+import com.applandeo.materialcalendarview.utils.CalendarProperties
+import com.applandeo.materialcalendarview.utils.isBetweenMinAndMax
+import java.util.*
+
+/**
+ * This class is responsible for handle click events
+ *
+ *
+ * Created by Applandeo Team.
+ */
+class DayRowLongClickListener(
+        private val calendarProperties: CalendarProperties
+) : OnItemLongClickListener {
+
+    override fun onItemLongClick(adapterView: AdapterView<*>, view: View, position: Int, id: Long): Boolean {
+        val day = GregorianCalendar().apply {
+            time = adapterView.getItemAtPosition(position) as Date
+        }
+
+        calendarProperties.onDayClickListener?.let {
+            onLongClick(day)
+        }
+
+        return true
+    }
+
+    private fun onLongClick(day: Calendar) {
+        if (calendarProperties.eventDays.isEmpty()) {
+            callOnLongClickListener(EventDay(day))
+            return
+        }
+
+        calendarProperties.eventDays.firstOrNull { it.calendar == day }.let {
+            callOnLongClickListener(it ?: EventDay(day))
+        }
+    }
```

Comment:
What is the purpose of scope function here ?

Reply:
"if" statement above can be removed because this scope function contains this logic inside